tests/test_rates.py

```diff
@@ -19,15 +31,49 @@
 )
 
 
-def test_token_rates():
+token_rates_testdata = [
+    ("gpt-4", "gpt-4", "0.00012"),
+    ("gpt-4", "gpt-35-turbo", "0.00012"),
+    ("gpt-4", "non-existent", "0.00012"),
+    ("gpt-4", "gpt-4", "0.00012"),
+    ("gpt-35-turbo", "gpt-4", "0.0000050"),
+    ("non-existent", "gpt-4", "0.00012"),
+    ("gpt-35-turbo", "gpt-4", "0.0000050"),
+    ("gpt-35-turbo", "gpt-35-turbo", "0.0000050"),
+    ("non-existent", "non-existent", "0"),
+]
+
+
+@pytest.mark.parametrize("deployment, model, price", token_rates_testdata)
+def test_token_rates(deployment: str, model: str, price: str):
     rates = RatesCalculator(MODEL_RATES)
-    price = rates.calculate_price(
-        "gpt-4", "hello", "hello", {"prompt_tokens": 2, "completion_tokens": 1}
+    calculated_price = rates.calculate_price(
+        deployment,
+        model,
+        "hello",
+        "hello",
+        {"prompt_tokens": 2, "completion_tokens": 1},
     )
-    assert price == Decimal("0.00012")
+    assert calculated_price == Decimal(price)
+
 
+token_rates_testdata = [
```

Comment:
You are redefining the same variable. Does it work correctly?

Reply:
Yes, it works. But I renamed for readability.